library/std/src/io/stdio.rs

```diff
@@ -712,26 +705,38 @@ impl Write for &Stdout {
     }
 }
 
+impl StdoutLock<'_> {
+    #[inline]
+    fn inner(&mut self) -> RefMut<'_, LineWriter<StdoutRaw>> {
+        #[cold]
+        fn init() -> LineWriter<StdoutRaw> {
+            LineWriter::new(stdout_raw())
+        }
+
+        RefMut::map(self.inner.borrow_mut(), |w| w.get_or_insert_with(init))
+    }
+}
+
 #[stable(feature = "rust1", since = "1.0.0")]
 impl Write for StdoutLock<'_> {
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
-        self.inner.borrow_mut().write(buf)
+        self.inner().write(buf)
     }
     fn write_vectored(&mut self, bufs: &[IoSlice<'_>]) -> io::Result<usize> {
-        self.inner.borrow_mut().write_vectored(bufs)
+        self.inner().write_vectored(bufs)
     }
     #[inline]
     fn is_write_vectored(&self) -> bool {
-        self.inner.borrow_mut().is_write_vectored()
+        stdout_raw().is_write_vectored()
```

Comment:
This is still using the raw stdout for this.

Reply:
Ah yes, I saw your comment too late... it's fixed now.